### PR TITLE
propogate and set the socket timeout on the connection

### PR DESCRIPTION
**Pull request by rphillips**
* Add test for memory benchmark, although it will be commented out
* Add an extra test for make sure a server that does not do a accept() does not leak

### CODE REVIEW

File: client/upgrade.lua [hunk 1]
```diff
@@ -72,7 +72,6 @@ local function loadWindowsFFI()
     typedef const CHAR *LPCSTR, *PCSTR;
 
     enum {
-    ERROR_SUCCESS = 0L,
```
Comment (rphillips) [inline client/upgrade.lua]: doubly defined

Comment (jjbuchan) [inline client/upgrade.lua]: Where else is it defined?  I can't see it.

Comment (rphillips) [inline client/upgrade.lua]: It's defined by the libc headers usually. I was getting a warning of it being doubly defined.

**Comment by jjbuchan** [comment]
+1

**Pull request merged**